internal/authtoken/repository.go

```diff
@@ -337,3 +338,19 @@ func (r *Repository) DeleteAuthToken(ctx context.Context, id string, opt ...Opti
 
 	return rowsDeleted, nil
 }
+
+// CreatePendingAuthToken creates a "pending" token in the repository using
+// the tokenRequestId as it's PublicId. The provided IAM User ID must be
+// associated to the provided auth account id or an error will be returned.  All
+// options are ignored.
+func (r *Repository) CreatePendingAuthToken(ctx context.Context, tokenRequestId string, iamUser *iam.User, withAuthAccountId string, opt ...Option) error {
+	panic("to-do")
+}
+
+// IssueAuthToken will retrieve the "pending" token and update it's status to
+// "issued".  If the token has already been issued, an error is returned with a
+// nil token.  If no token is found for the tokenRequestId an error is returned
+// with a nil token.
+func (r *Repository) IssueAuthToken(ctx context.Context, tokenRequestId string) (*AuthToken, error) {
```

Comment:
Can we discuss the flow of these methods?

Reply:
Happy to.